automates/model_assembly/networks.py

```diff
@@ -1057,16 +1112,17 @@ def _(
             live_variables: Dict[VariableIdentifier, VariableNode],
```

Comment:
does live variables still need to be passed around?

Reply:
We are still using `live_variables` to keep track of all the nodes included under a GrFNSubgraph:
```python
con_subgraph.nodes.extend(list(live_variables.values()))
```

We use those nodes in the `__call__` method of the GrFNSubgraph so I definitely think we need to keep the attr around and therefore we still need to use `live_variables`:
```python
if executed:
    all_nodes_visited.update(executed_visited_variables)
    all_nodes_visited.add(node_to_execute)
    node_execute_queue.extend(
        [
            succ
            for var in executed_visited_variables
            for succ in grfn.successors(var)
            if (
                succ in self.nodes
                and succ not in all_nodes_visited
            )
            or (
                var in self.nodes
                and succ.func_type == LambdaType.INTERFACE
            )
        ]
    )
else:
    node_execute_queue.extend(
        [
            lambda_pred
            for var_pred in grfn.predecessors(node_to_execute)
            for lambda_pred in grfn.predecessors(var_pred)
            if (
                lambda_pred in self.nodes
                and lambda_pred not in all_nodes_visited
            )
            or lambda_pred.func_type == LambdaType.INTERFACE
        ]
    )
    node_execute_queue.append(node_to_execute)
```